test/unit/test_gateway.c

```diff
@@ -1053,6 +1055,47 @@ TEST_CASE(exec, barrier_closing, NULL)
 	return MUNIT_OK;
 }
 
+/* Send an exec request in the new (schema version 1) format, which
+ * supports larger numbers of parameters. */
+TEST_CASE(exec, manyParams, NULL)
+{
+	struct exec_fixture *f = data;
+	uint64_t stmt_id;
+	size_t len = 20000;
+	char *sql = munit_malloc(len);
+	size_t pos;
+	size_t i;
+	size_t num_exec_params = 999;
+	struct value *values = munit_calloc(num_exec_params, sizeof(*values));
+	(void)params;
+
+	pos = snprintf(sql, len, "SELECT (n) FROM test WHERE n = ?");
```

Comment:
I would expect a `SELECT` statement when testing `query` not `exec`. I'm not sure if it makes a whole lot of difference in this case, but would change it for consistency with the other tests.

Reply:
Updated to use `DELETE FROM` instead